app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ingredients/IngredientsProductFragment.java

```diff
@@ -290,7 +290,7 @@ public void refreshView(State state) {
             substanceProduct.append(" ");
 
             String allergen;
-            for (int i = 0; i < allergens.size() - 1; i++) {
+            for (int i = 0; i <= allergens.size() - 1; i++) {
```

Comment:
This should actually read as the following `for (int i = 0; i < allergens.size(); i++)`

Reply:
@Karljoones I'll change the loop condition.